pos_disable_pricelist_selection/models/pos_config.py

```diff
@@ -1,9 +1,75 @@
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class PosConfig(models.Model):
     _inherit = "pos.config"
 
+    def _default_pricelist(self):
+        return self.env["product.pricelist"].search(
+            [
+                ("company_id", "in", (False, self.env.company.id)),
+                ("currency_id", "=", self.env.company.currency_id.id),
+            ],
+            limit=1,
+        )
+
     hide_pricelist_button = fields.Boolean(
         default=False,
     )
+    selectable_pricelist_ids = fields.Many2many(
+        "product.pricelist",
+        string="Selectable Pricelists",
+        domain="[('id', 'in', available_pricelist_ids)]",
+        relation="pos_conf_selectable_pricelist_rel",
+        default=_default_pricelist,
+    )
+    pricelist_id_domain = fields.Binary(
+        compute="_compute_pricelist_id_domain",
+        readonly=True,
+        store=False,
+    )
+
+    @api.depends(
+        "hide_pricelist_button", "allowed_pricelist_ids", "selectable_pricelist_ids"
+    )
+    def _compute_pricelist_id_domain(self):
+        for rec in self:
+            if rec.hide_pricelist_button:
+                rec.pricelist_id_domain = [("id", "in", rec.allowed_pricelist_ids.ids)]
+            else:
+                rec.pricelist_id_domain = [
+                    ("id", "in", rec.selectable_pricelist_ids.ids)
+                ]
```

Comment:
**chore**: Documentation of the `web_domain_field` recommends using `json.dumps` to make sure any edge case is handled properly.

Reply:
Couldn't find a way to cause an error, so considering this non-blocking.